src/services/LiqualityService.ts

```diff
@@ -0,0 +1,113 @@
+import * as bitcoin from 'bitcoinjs-lib';
+import {
+  LiqualityAddress,
+  LiqualityGetAddressesResponse,
+  LiqualityMethods, LiqualitySignedTx,
+  LiqualityTx,
+  WalletAddress,
+  WindowBitcoinProvider,
+} from '@/types';
+import { WalletService } from '@/services';
+import { EnvironmentAccessorService } from '@/services/enviroment-accessor.service';
+
+export default class LiqualityService extends WalletService {
+  private bitcoinProvider!: WindowBitcoinProvider;
+
+  private enabled = false;
+
+  constructor(testBitcoinProvider?: WindowBitcoinProvider) {
+    super();
+    if (testBitcoinProvider) {
+      this.bitcoinProvider = testBitcoinProvider;
+      this.enabled = true;
+    }
+  }
+
+  private enable(): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      const w = window as Window;
+      if (!this.enabled) {
+        try {
+          if (w.bitcoin) {
+            this.bitcoinProvider = window.bitcoin;
+            this.bitcoinProvider.enable()
+              .then(([address]: LiqualityAddress[]) => {
+                this.enabled = !!address;
+                resolve();
+              });
+          }
+        } catch (e) {
+          reject(e);
+        }
+      } else {
+        resolve();
+      }
+    });
+  }
+
+  getAccountAddresses(batch: number, index: number): Promise<WalletAddress[]> {
+    return new Promise<WalletAddress[]>((resolve, reject) => {
+      const walletAddresses: WalletAddress[] = [];
+      this.enable()
+        .then(() => Promise.all([
+          this.bitcoinProvider.request({
+            method: LiqualityMethods.GET_ADDRESS,
+            params: [index, batch, true],
+          }),
+          this.bitcoinProvider.request({
+            method: LiqualityMethods.GET_ADDRESS,
+            params: [index, batch, false],
+          }),
+        ]))
+        .then(([changeAddreses, noChangeAddresses]) => {
+          const addresses = noChangeAddresses as LiqualityGetAddressesResponse[];
+          addresses.concat(changeAddreses as LiqualityGetAddressesResponse[])
+            .forEach((liqualityAddress: LiqualityGetAddressesResponse) => {
+              walletAddresses.push({
+                address: liqualityAddress.address,
+                serializedPath: liqualityAddress.derivationPath,
+                publicKey: liqualityAddress.publicKey,
+                path: [0],
```

Comment:
what's this path? how does it differ with serializedPath?

Reply:
Its the same. `WalletAddress` was created with trezor returned names `84'/1'/0'/0/1` is serializedPath for trezor and derivationPath for Liquality